apps/wallet/src/ui/app/pages/home/tokens/TokensDetails.tsx

```diff
@@ -57,7 +62,110 @@ function PinButton({ unpin, onClick }: { unpin?: boolean; onClick: () => void })
 	);
 }
 
-function MyTokens({
+function TokenRowButton({
+	coinBalance,
+	page,
+	copy,
+}: {
+	coinBalance: CoinBalanceType;
+	page: 'send' | 'swap';
+	copy: string;
+}) {
+	const params = new URLSearchParams({
+		type: coinBalance.coinType,
+	});
+
+	return (
+		<ButtonOrLink
+			to={`/${page}?${params.toString()}`}
+			onClick={
+				page === 'send'
+					? () =>
+							ampli.selectedCoin({
+								coinType: coinBalance.coinType,
+								totalBalance: Number(BigInt(coinBalance.totalBalance) / MIST_PER_SUI),
+							})
+					: undefined
+			}
+			key={coinBalance.coinType}
+			className="no-underline text-subtitle font-medium text-steel hover:font-semibold hover:text-hero"
+		>
+			{copy}
+		</ButtonOrLink>
+	);
+}
+
+export function TokenRow({
+	coinBalance,
+	renderActions,
+	onClick,
+	as: Tag = 'div',
+	borderBottom,
+}: {
+	coinBalance: CoinBalanceType;
+	renderActions?: boolean;
+	as?: 'div' | 'button';
+	onClick?: () => void;
+	borderBottom?: boolean;
```

Comment:
I hate props like these, I generally like these dividers to be in the consuming parents, it almost always makes more sense there (e.g. divide-y, manually adding divider nodes, wrapper divs)